src/main/java/net/dries007/tfc/objects/te/TEFirePit.java

```diff
@@ -125,6 +125,15 @@ public void update()
             IBlockState state = world.getBlockState(pos);
             if (state.getValue(LIT))
             {
+                // Have to check the above block, since minecraft think this block is "roof"
+                if (world.isRainingAt(pos.up()))
+                {
+                    // Instantly consume last fuel and turn off
+                    burnTicks = 0;
+                    burnTemperature = 0;
+                    world.setBlockState(pos, state.withProperty(LIT, false));
+                    return;
+                }
```

Comment:
Same as above, is there a reason this isn't done on random tick? That would make more sense thematically.